### ISSUE TITLE
Spaceship.login fail sometimes

### ISSUE THREAD

**Issue by neil-wu** [fastlane/fastlane#4240]
Spaceship.login fail in my mac. But it was OK when running in another mac in the same network. 

fastlane version is 1.81.0 

##### Complete output

```
/Users/appdev/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/net/http/response.rb:266:in `raise': exception object expected (TypeError)
	from /Users/appdev/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/net/http/response.rb:266:in `rescue in ensure in inflater'
	from /Users/appdev/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/net/http/response.rb:263:in `ensure in inflater'
	from /Users/appdev/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/net/http/response.rb:263:in `inflater'
	from /Users/appdev/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/net/http/response.rb:279:in `read_body_0'
	from /Users/appdev/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/net/http/response.rb:201:in `read_body'
	from /Users/appdev/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/net/http.rb:1134:in `block in get'
	from /Users/appdev/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/net/http.rb:1421:in `block (2 levels) in transport_request'
	from /Users/appdev/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/net/http/response.rb:162:in `reading_body'
	from /Users/appdev/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/net/http.rb:1420:in `block in transport_request'
	from /Users/appdev/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/net/http.rb:1411:in `catch'
	from /Users/appdev/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/net/http.rb:1411:in `transport_request'
	from /Users/appdev/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/net/http.rb:1384:in `request'
	from /Users/appdev/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/net/http.rb:1377:in `block in request'
	from /Users/appdev/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/net/http.rb:853:in `start'
	from /Users/appdev/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/net/http.rb:1375:in `request'
	from /Users/appdev/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/net/http.rb:1133:in `get'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/faraday-0.9.2/lib/faraday/adapter/net_http.rb:80:in `perform_request'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/faraday-0.9.2/lib/faraday/adapter/net_http.rb:40:in `block in call'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/faraday-0.9.2/lib/faraday/adapter/net_http.rb:87:in `with_net_http_connection'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/faraday-0.9.2/lib/faraday/adapter/net_http.rb:32:in `call'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/faraday-cookie_jar-0.0.6/lib/faraday/cookie_jar.rb:22:in `call'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/faraday_middleware-0.10.0/lib/faraday_middleware/response_middleware.rb:30:in `call'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/faraday_middleware-0.10.0/lib/faraday_middleware/response_middleware.rb:30:in `call'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/faraday_middleware-0.10.0/lib/faraday_middleware/response_middleware.rb:30:in `call'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/faraday-0.9.2/lib/faraday/rack_builder.rb:139:in `build_response'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/faraday-0.9.2/lib/faraday/connection.rb:377:in `run_request'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/faraday-0.9.2/lib/faraday/connection.rb:140:in `get'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/spaceship-0.25.1/lib/spaceship/client.rb:404:in `block in send_request'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/spaceship-0.25.1/lib/spaceship/client.rb:322:in `with_retry'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/spaceship-0.25.1/lib/spaceship/client.rb:403:in `send_request'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/spaceship-0.25.1/lib/spaceship/client.rb:379:in `request'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/spaceship-0.25.1/lib/spaceship/client.rb:313:in `itc_service_key'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/spaceship-0.25.1/lib/spaceship/client.rb:274:in `block in send_shared_login_request'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/faraday-0.9.2/lib/faraday/connection.rb:374:in `block in run_request'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/faraday-0.9.2/lib/faraday/connection.rb:388:in `block in build_request'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/faraday-0.9.2/lib/faraday/request.rb:26:in `block in create'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/faraday-0.9.2/lib/faraday/request.rb:25:in `tap'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/faraday-0.9.2/lib/faraday/request.rb:25:in `create'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/faraday-0.9.2/lib/faraday/connection.rb:384:in `build_request'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/faraday-0.9.2/lib/faraday/connection.rb:370:in `run_request'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/faraday-0.9.2/lib/faraday/connection.rb:177:in `post'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/spaceship-0.25.1/lib/spaceship/client.rb:404:in `block in send_request'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/spaceship-0.25.1/lib/spaceship/client.rb:322:in `with_retry'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/spaceship-0.25.1/lib/spaceship/client.rb:403:in `send_request'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/spaceship-0.25.1/lib/spaceship/client.rb:379:in `request'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/spaceship-0.25.1/lib/spaceship/client.rb:273:in `send_shared_login_request'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/spaceship-0.25.1/lib/spaceship/portal/portal_client.rb:35:in `send_login_request'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/spaceship-0.25.1/lib/spaceship/client.rb:346:in `do_login'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/spaceship-0.25.1/lib/spaceship/client.rb:225:in `login'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/spaceship-0.25.1/lib/spaceship/client.rb:88:in `login'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/spaceship-0.25.1/lib/spaceship/portal/spaceship.rb:22:in `login'
	from /Users/appdev/.rvm/gems/ruby-2.2.2@global/gems/spaceship-0.25.1/lib/spaceship/portal/spaceship.rb:77:in `login'
	from check_device_num.rb:10:in `<main>'
```

**Comment by TKBurner** [comment]
Hi @neil-wu Thanks for reaching out. Can you share the full information requested in the issue template? https://github.com/fastlane/fastlane/issues/new Thanks! 👍

**Comment by mfurtak** [comment]
Hi @neil-wu  :wave: - we just released [`spaceship` 0.26.3](https://github.com/fastlane/fastlane/releases/tag/spaceship/0.26.3) 🚀  that includes an experimental workaround behind a feature switch.

Could you please upgrade your `spaceship` with:

```bash
gem update spaceship
```

and make sure that 

```bash
gem which spaceship
```

returns a path including `0.26.3`, and then try running:

```bash
SPACESHIP_LOGIN_ENCODING_IDENTITY=1 fastlane [YOUR PARAMS]
```

or whatever other command you are using, and let me know if it allows you to succeed? Thank you very much!

**Comment by mfurtak** [comment]
Hi all, we've released [`spaceship` 0.27.1](https://github.com/fastlane/fastlane/releases/tag/spaceship/0.27.1) 🚀  This makes the fix provided by `SPACESHIP_LOGIN_ENCODING_IDENTITY` enabled by default, so if you upgrade, you should no longer need to use that environment variable.

Thanks for your help in investigating this issue! 👍
